### ISSUE TITLE
Tests fail on Windows

### ISSUE THREAD

**Issue by hankache** [Leont/path-finder#6]
Rakudo 2019.07.1
```
===> Testing: Path::Finder:ver<0.2.0>
# Failed test 'Depth first iteration (pre)'
# at t/breadth-depth-files.t line 41
# expected: $["aaaa.txt", "bbbb.txt", "cccc/dddd.txt", "cccc/eeee/ffff.txt", "gggg.txt"]
#      got: $["aaaa.txt", "bbbb.txt", "cccc\\dddd.txt", "cccc\\eeee\\ffff.txt", "gggg.txt"]
# ["aaaa.txt", "bbbb.txt", "cccc\\dddd.txt", "cccc\\eeee\\ffff.txt", "gggg.txt"]
# Failed test 'Depth first iteration (post)'
# at t/breadth-depth-files.t line 44
# expected: $["aaaa.txt", "bbbb.txt", "cccc/dddd.txt", "cccc/eeee/ffff.txt", "gggg.txt"]
#      got: $["aaaa.txt", "bbbb.txt", "cccc\\dddd.txt", "cccc\\eeee\\ffff.txt", "gggg.txt"]
# ["aaaa.txt", "bbbb.txt", "cccc\\dddd.txt", "cccc\\eeee\\ffff.txt", "gggg.txt"]
# You failed 2 tests of 2
# Failed test 'Breadth first iteration'
# at t/breadth-depth.t line 58
# expected: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "cccc/dddd.txt", "cccc/eeee", "cccc/eeee/ffff.txt"]
#      got: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "cccc\\dddd.txt", "cccc\\eeee", "cccc\\eeee\\ffff.txt"]
# Failed test 'Depth first iteration (pre)'
# at t/breadth-depth.t line 61
# expected: $[".", "aaaa.txt", "bbbb.txt", "cccc", "cccc/dddd.txt", "cccc/eeee", "cccc/eeee/ffff.txt", "gggg.txt"]
#      got: $[".", "aaaa.txt", "bbbb.txt", "cccc", "cccc\\dddd.txt", "cccc\\eeee", "cccc\\eeee\\ffff.txt", "gggg.txt"]
# Failed test 'Depth first iteration (post)'
# at t/breadth-depth.t line 64
# expected: $["aaaa.txt", "bbbb.txt", "cccc/dddd.txt", "cccc/eeee/ffff.txt", "cccc/eeee", "cccc", "gggg.txt", "."]
#      got: $["aaaa.txt", "bbbb.txt", "cccc\\dddd.txt", "cccc\\eeee\\ffff.txt", "cccc\\eeee", "cccc", "gggg.txt", "."]
# You failed 3 tests of 3
# Failed test 'skip() with custom rule'
# at t/logic.t line 48
# expected: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "cccc/dddd.txt"]
#      got: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "cccc\\dddd.txt"]
# Failed test 'skip() with crazy custom rule'
# at t/logic.t line 62
# expected: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "cccc/dddd.txt"]
#      got: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "cccc\\dddd.txt"]
# Failed test 'skip() with skip'
# at t/logic.t line 75
# expected: $[".", "aaaa.txt", "bbbb.txt", "cccc", "cccc/dddd.txt"]
#      got: $[".", "aaaa.txt", "bbbb.txt", "cccc", "cccc\\dddd.txt"]
# Failed test 'none with references'
# at t/logic.t line 93
# expected: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "cccc/dddd.txt"]
#      got: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "cccc\\dddd.txt"]
# Failed test 'nested and + or with prunning'
# at t/logic.t line 103
# expected: $["cccc/eeee"]
#      got: $["cccc\\eeee"]
# You failed 5 tests of 7
# Failed test 'min_depth(3) test'
# at t/min-max-depth.t line 29
# expected: $["cccc/eeee/ffff.txt", "hhhh/iiii/jjjj/kkkk/llll/mmmm.txt"]
#      got: $["cccc\\eeee\\ffff.txt", "hhhh\\iiii\\jjjj\\kkkk\\llll\\mmmm.txt"]
# Failed test 'max_depth(2) test'
# at t/min-max-depth.t line 41
# expected: $["aaaa.txt", "bbbb.txt", "gggg.txt", "cccc/dddd.txt"]
#      got: $["aaaa.txt", "bbbb.txt", "gggg.txt", "cccc\\dddd.txt"]
# Failed test 'min_depth(2)->max_depth(3) test'
# at t/min-max-depth.t line 51
# expected: $["cccc/dddd.txt", "cccc/eeee/ffff.txt"]
#      got: $["cccc\\dddd.txt", "cccc\\eeee\\ffff.txt"]
# You failed 3 tests of 3
# Failed test 'name('Foo.*') match'
# at t/names.t line 33
# expected: $["lib/Foo.pm", "lib/Foo.pod"]
#      got: $["lib\\Foo.pm", "lib\\Foo.pod"]
# Failed test 'name(qr/Foo/) match'
# at t/names.t line 43
# expected: $["lib/Foo.pm", "lib/Foo.pod"]
#      got: $["lib\\Foo.pm", "lib\\Foo.pod"]
# Failed test 'name(qr/Foo/) match'
# at t/names.t line 53
# expected: $["lib/Foo.pm", "lib/Foo.pod"]
#      got: $["lib\\Foo.pm", "lib\\Foo.pod"]
# Failed test 'name('*.pod', '*.pm') match'
# at t/names.t line 63
# expected: $["lib/Foo.pm", "lib/Foo.pod"]
#      got: $["lib\\Foo.pm", "lib\\Foo.pod"]
# You failed 4 tests of 5
# Failed test 'Depth first iteration (pre)'
# at t/relative.t line 47
# expected: $[".", "aaaa.txt", "bbbb.txt", "cccc", "cccc/dddd.txt", "cccc/eeee", "cccc/eeee/ffff.txt", "gggg.txt"]
#      got: $[".", "aaaa.txt", "bbbb.txt", "cccc", "cccc\\dddd.txt", "cccc\\eeee", "cccc\\eeee\\ffff.txt", "gggg.txt"]
# Failed test 'Depth first iteration (post)'
# at t/relative.t line 50
# expected: $["aaaa.txt", "bbbb.txt", "cccc/dddd.txt", "cccc/eeee/ffff.txt", "cccc/eeee", "cccc", "gggg.txt", "."]
#      got: $["aaaa.txt", "bbbb.txt", "cccc\\dddd.txt", "cccc\\eeee\\ffff.txt", "cccc\\eeee", "cccc", "gggg.txt", "."]
# You failed 2 tests of 2
# Failed test 'size > 0'
# at t/stat_tests.t line 25
# expected: 'C:\Users\naoum\AppData\Local\Temp\oQMlOPxWj3\data/Changes'
#      got: 'C:\Users\naoum\AppData\Local\Temp\oQMlOPxWj3\data\Changes'
# [IO::Path.new("C:\\Users\\naoum\\AppData\\Local\\Temp\\oQMlOPxWj3\\data\\Changes", :SPEC(IO::Spec::Win32), :CWD("C:\\Users\\naoum\\.zef\\store\\Path-Finder-0.2.0.tar.gz\\Path-Finder-0.2.0"))]
# You failed 1 test of 2
# Failed test 'Follow symlinks'
# at t/symlink.t line 79
# expected: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "pppp", "qqqq.txt", "cccc/dddd.txt", "cccc/eeee", "pppp/ffff.txt"]
#      got: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "pppp", "qqqq.txt", "cccc\\dddd.txt", "cccc\\eeee", "cccc\\eeee\\ffff.txt"]
# Failed test 'Follow symlinks, but loop_safe = 0'
# at t/symlink.t line 82
# expected: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "pppp", "qqqq.txt", "cccc/dddd.txt", "cccc/eeee", "pppp/ffff.txt", "cccc/eeee/ffff.txt"]
#      got: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "pppp", "qqqq.txt", "cccc\\dddd.txt", "cccc\\eeee", "cccc\\eeee\\ffff.txt"]
# Failed test 'Don't follow symlinks, but report them'
# at t/symlink.t line 85
# expected: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "pppp", "qqqq.txt", "cccc/dddd.txt", "cccc/eeee", "cccc/eeee/ffff.txt"]
#      got: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "pppp", "qqqq.txt", "cccc\\dddd.txt", "cccc\\eeee", "cccc\\eeee\\ffff.txt"]
# Failed test 'Don't follow or report symlinks'
# at t/symlink.t line 88
# expected: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "cccc/dddd.txt", "cccc/eeee", "cccc/eeee/ffff.txt"]
#      got: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "cccc\\dddd.txt", "cccc\\eeee", "cccc\\eeee\\ffff.txt"]
# Failed test 'No dangling symlinks'
# at t/symlink.t line 127
# expected: $[".", "aaaa.txt", "bbbb.txt", "cccc", "qqqq.txt", "cccc/dddd.txt"]
#      got: $[".", "aaaa.txt", "bbbb.txt", "cccc", "qqqq.txt", "cccc\\dddd.txt"]
# Failed test 'Symlink loop'
# at t/symlink.t line 157
# expected: $[".", "aaaa.txt", "bbbb.txt", "cccc", "cccc/dddd.txt", "cccc/eeee"]
#      got: $[".", "aaaa.txt", "bbbb.txt", "cccc", "cccc\\dddd.txt", "cccc\\eeee"]
# You failed 6 tests of 8
# Failed test 'not_vcs test'
# at t/vcs.t line 34
# expected: $["aaaa.txt", "bbbb.txt"]
#      got: $["aaaa.txt", "bbbb.txt", "cccc\\.bzr\\foo", "cccc\\.git\\foo", "cccc\\.hg\\foo", "cccc\\.svn\\foo", "cccc\\CVS\\foo", "cccc\\RCS\\foo", "cccc\\_darcs\\foo"]
# You failed 1 test of 1
===> Testing [FAIL]: Path::Finder:ver<0.2.0>
Aborting due to test failure: Path::Finder:ver<0.2.0> (use --force-test to override)
```

**Comment by Leont** [comment]
I just released 0.2.1, which should fix this

**Comment by Leont** [comment]
Actually, a closer inspection of the output reveals two more issues I had initially missed. The first (in `t/stat_tests.t` is fixes easily enough. The `t/vcs.t` failure is still a bit a mystery to me though.

**Comment by hankache** [comment]
@Leont ++

**Comment by Leont** [comment]
I figured out the second issue with some help from @araraloren, now released 0.2.2 :-)

**Comment by hankache** [comment]
@Leont thanks.
Unfortunately some tests are still failing:
```
===> Testing: Path::Finder:ver<0.2.2>
# Failed test 'Depth first iteration (pre)'
# at t/relative.t line 47
# expected: $[".", "aaaa.txt", "bbbb.txt", "cccc", "cccc/dddd.txt", "cccc/eeee", "cccc/eeee/ffff.txt", "gggg.txt"]
#      got: $[".", "aaaa.txt", "bbbb.txt", "cccc", "cccc\\dddd.txt", "cccc\\eeee", "cccc\\eeee\\ffff.txt", "gggg.txt"]
# Failed test 'Depth first iteration (post)'
# at t/relative.t line 50
# expected: $["aaaa.txt", "bbbb.txt", "cccc/dddd.txt", "cccc/eeee/ffff.txt", "cccc/eeee", "cccc", "gggg.txt", "."]
#      got: $["aaaa.txt", "bbbb.txt", "cccc\\dddd.txt", "cccc\\eeee\\ffff.txt", "cccc\\eeee", "cccc", "gggg.txt", "."]
# You failed 2 tests of 2
# Failed test 'Follow symlinks'
# at t/symlink.t line 79
# expected: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "pppp", "qqqq.txt", "cccc/dddd.txt", "cccc/eeee", "pppp/ffff.txt"]
#      got: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "pppp", "qqqq.txt", "cccc/dddd.txt", "cccc/eeee", "cccc/eeee/ffff.txt"]
# Failed test 'Follow symlinks, but loop_safe = 0'
# at t/symlink.t line 82
# expected: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "pppp", "qqqq.txt", "cccc/dddd.txt", "cccc/eeee", "pppp/ffff.txt", "cccc/eeee/ffff.txt"]
#      got: $[".", "aaaa.txt", "bbbb.txt", "cccc", "gggg.txt", "pppp", "qqqq.txt", "cccc/dddd.txt", "cccc/eeee", "cccc/eeee/ffff.txt"]
# You failed 2 tests of 8
===> Testing [FAIL]: Path::Finder:ver<0.2.2>
Aborting due to test failure: Path::Finder:ver<0.2.2> (use --force-test to override)
```

**Comment by Leont** [comment]
I think we can safely skip the symlink test on windows, and I think I just fixed the relative paths tests as well. Just released 0.2.3
